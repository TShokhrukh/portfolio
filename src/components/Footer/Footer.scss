.Footer {
  --footer-background-color: #{rgba($purple, 0.1)};
  --footer-padding-y: #{$spacer};
  --footer-padding-x: 0;

  background-color: var(--footer-background-color);
  padding: var(--footer-padding-y) var(--footer-padding-x);
}

.Footer-Content {
  display: flex;
  justify-content: center;
}
