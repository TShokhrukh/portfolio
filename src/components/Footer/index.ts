export * from './build'
export * from './__types__'
export { Footer as FooterBase } from './Footer'
