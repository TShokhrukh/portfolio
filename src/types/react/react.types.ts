import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
