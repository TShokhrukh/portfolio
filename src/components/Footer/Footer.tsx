import React, { FC } from 'react'
import { Icon } from '~/components/Icon'
import { FooterSocialLink } from './SocialLink'
import { FooterContent } from './Content'
import { cnFooter } from './Footer.const'
import { environment } from '~/environment'
import type { IFooterProps } from './__types__'
import './Footer.scss'

export const Footer: FC<IFooterProps> = ({
  component: Component = 'footer',
  className,
  ...props
}) => {
  return (
    <Component {...props} className={cnFooter({}, [className])}>
      <FooterContent>
        <FooterSocialLink href={environment.contacts.git} target='_blank' rel='noopener noreferrer'>
          <Icon name='github' />
        </FooterSocialLink>
      </FooterContent>
    </Component>
  )
}
