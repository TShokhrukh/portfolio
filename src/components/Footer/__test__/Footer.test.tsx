/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { Footer } from '../Footer'

describe('Footer', () => {
  test('component template', () => {
    render(<Footer className='test_class' data-testid='footer_id' />)

    expect(screen.getByTestId('footer_id')).toMatchSnapshot()
  })
})
