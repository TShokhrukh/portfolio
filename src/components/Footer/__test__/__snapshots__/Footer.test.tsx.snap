// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component template 1`] = `
<footer
  class="Footer test_class"
  data-testid="footer_id"
>
  <div
    class="Container Footer-Content"
  >
    <a
      class="Link Footer-SocialLink"
      href="https://github.com/TShokhrukh"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        aria-hidden="true"
        class="fa fa-github Icon"
      />
    </a>
  </div>
</footer>
`;
